addon/-private/system/references/has-many.js

```diff
@@ -46,10 +150,85 @@ HasManyReference.prototype.ids = function() {
   });
 };
 
+/**
+   The link Ember Data will use to fetch or reload this belongs-to
+   relationship.
+
+   Example
+
+   ```javascript
+   // models/post.js
+   export default DS.Model.extend({
+     comments: DS.hasMany({ async: true })
+   });
+
+   var post = store.push({
+     type: 'post',
+     id: 1,
+     relationships: {
+       comments: {
+         links: {
+           related: {
+             href: '/posts/1/comments',
+             meta: {
+               count: 10
+             }
+         }
+       }
+     }
+   });
+
+   var commentsRef = post.hasMany('comments');
+
+   commentsRef.meta(); // { count: 10 }
+   ```
+
+   @method meta
+   @return {Object} The meta information for the has-many relationship.
+*/
 HasManyReference.prototype.meta = function() {
   return this.hasManyRelationship.meta;
 };
 
+/**
+   `push` can be used to update the data in the relationship and Ember
+   Data will treat the new data as the conanical value of this
+   relationship on the backend.
+
+   Example
+
+   ```javascript
+   // models/post.js
+   export default DS.Model.extend({
+     comments: DS.hasMany({ async: true })
+   });
+
+   var post = store.push({
+     type: 'post',
+     id: 1,
+     relationships: {
+       comments: {
+         data: [{ type: 'comment', id: 1 }]
+       }
+     }
+   });
+
+   var commentsRef = post.hasMany('comments');
+
+   commentsRef.ids(); // ['1']
+
+   commentsRef.push([
```

Comment:
I am sad that this format has been added 😔, but #4398  and #4588 have been opened to address this.

Not sure if we want to advertise this in the meantime, until those PRs are in stable...

What do you think?